charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml

```diff
@@ -253,13 +253,13 @@ spec:
   scrapeConfigSelector:
     matchLabels:
       release: {{ $.Release.Name | quote }}
-{{ else }}
+{{ else if not (eq .Values.prometheus.prometheusSpec.scrapeConfigSelector nil) }}
   scrapeConfigSelector: {}
 {{- end }}
 {{- if .Values.prometheus.prometheusSpec.scrapeConfigNamespaceSelector }}
   scrapeConfigNamespaceSelector:
 {{ tpl (toYaml .Values.prometheus.prometheusSpec.scrapeConfigNamespaceSelector | indent 4) . }}
-{{ else }}
+{{ else if not (eq .Values.prometheus.prometheusSpec.scrapeConfigNamespaceSelector nil) }}
```

Comment:
what not `ne` instead of `not (eq` ?

Reply:
Good point @jkroepke, that's shorter and easier to read. I added unit test and they validate that both expressions achieve the same result.
Also added docs.

Should we implement this behavior for the same selectors as the original PR?
In my use case only the ones included in the PR caused the problem.